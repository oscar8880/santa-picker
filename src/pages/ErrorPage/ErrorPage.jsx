import styles from './ErrorPage.module.scss';
import React from 'react';
import Text from '../../components/Text/Text';
import Action from '../../components/Action/Action';
import PageWrapper from '../../components/PageWrapper/PageWrapper';
import { Container, Row, Col } from 'react-grid-system';
import chimneySantaImage from '../../images/chimneySanta.png';

const ErrorPage = () => {
  return (
    <div className={styles.ErrorPage}>
      <PageWrapper>
        <Container fluid>
          <Row>
            <Col xs={12} md={6} lg={6}>
              <div className={styles.ErrorPage__StrapLine}>
                <Text appearance="h6" tagName="h1">
                  Oops!
                </Text>
              </div>
              <ul className={styles.ErrorPage__BlurbTextList}>
                <li className={styles.ErrorPage__BlurbTextListItem}>
                  <Text appearance="h5">
                    Santa got stuck along the way and wasn&apos;t able to
                    deliver the emails
                  </Text>
                </li>
                <li className={styles.ErrorPage__BlurbTextListItem}>
                  <Text appearance="h5">
                    There&apos;s a good chance he&apos;s already used up his
                    email quota
                  </Text>
                </li>
                <li className={styles.ErrorPage__BlurbTextListItem}>
                  <Text appearance="h5">Please try again</Text>
                </li>
              </ul>
              <div className={styles.ErrorPage__CTA}>
                <Action tagName="Link" to="/" appearance="button-secondary">
                  Home
                </Action>
              </div>
            </Col>
            <Col xs={12} md={6} lg={5} offset={{ lg: 1 }}>
              <div className={styles.ErrorPage__Image}>
                <img src={chimneySantaImage} alt="Sleigh" width="80%"></img>
              </div>
            </Col>
          </Row>
        </Container>
      </PageWrapper>
    </div>
  );
};

export default ErrorPage;
