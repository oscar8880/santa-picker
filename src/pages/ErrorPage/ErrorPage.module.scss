@import "../../scss/functions";
@import "../../scss/variables";

.ErrorPage {
  display: flex;
  flex-direction: column;
  padding-top: $spacer-10;
  @media (min-width: $breakpoint-md) {
    padding-bottom: $spacer-12;
    padding-top: $spacer-12;
  }
}

.ErrorPage__StrapLine {
  padding-bottom: $spacer-8;
  text-align: center;
  @media (min-width: $breakpoint-md) {
    text-align: left;
  }
}

.ErrorPage__BlurbTextList {
  list-style: none;
  padding-bottom: $spacer-4;
  text-align: center;
  @media (min-width: $breakpoint-md) {
    text-align: left;
  }
}

.ErrorPage__BlurbTextListItem {
  color: $grey-500;
  padding-bottom: $spacer-6;
}

.ErrorPage__CTA {
  display: flex;
  justify-content: center;
  padding-bottom: $spacer-8;

  @media (min-width: $breakpoint-md) {
    justify-content: flex-start;
  }
}

.ErrorPage__Image {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  padding-top: $spacer-8;
  width: 80%;

  @media (min-width: $breakpoint-md) {
    justify-content: flex-start;
    padding-top: $spacer-9;
    width: 100%;
  }
}
