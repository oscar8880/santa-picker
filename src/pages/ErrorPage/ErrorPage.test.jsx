import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import ErrorPage from './ErrorPage';

test('matches snapshot', async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <ErrorPage />
    </MemoryRouter>,
  );

  expect(asFragment()).toMatchSnapshot();
});
