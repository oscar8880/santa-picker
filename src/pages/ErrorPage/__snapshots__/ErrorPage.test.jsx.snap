// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ErrorPage"
  >
    <div
      class="PageWrapper"
    >
      <div
        style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px;"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
          >
            <div
              class="ErrorPage__StrapLine"
            >
              <h1
                class="Text Text__h6"
              >
                Oops!
              </h1>
            </div>
            <ul
              class="ErrorPage__BlurbTextList"
            >
              <li
                class="ErrorPage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h5"
                >
                  Santa got stuck along the way and wasn't able to deliver the emails
                </p>
              </li>
              <li
                class="ErrorPage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h5"
                >
                  There's a good chance he's already used up his email quota
                </p>
              </li>
              <li
                class="ErrorPage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h5"
                >
                  Please try again
                </p>
              </li>
            </ul>
            <div
              class="ErrorPage__CTA"
            >
              <a
                class="Action Action__Appearance--button-secondary"
                href="/"
              >
                Home
              </a>
            </div>
          </div>
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 41.66666666666667%; flex-basis: 41.66666666666667%; flex-grow: 0; flex-shrink: 0; max-width: 41.66666666666667%; margin-left: 8.333333333333334%;"
          >
            <div
              class="ErrorPage__Image"
            >
              <img
                alt="Sleigh"
                src="chimneySanta.png"
                width="80%"
              />
            </div>
          </div>
        </div>
        <span
          style="display: table; clear: both;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
